rust/index/src/fulltext/util.rs

```diff
@@ -0,0 +1,144 @@
+/// A token instance is a unique value containing a trigram, an offset ID, and optionally a position within a document.
+/// These three attributes are packed into a single u128 value:
+/// - The trigram is a 63-bit value, packed into the top 64 bits.
```

Comment:
worth explaining why this is OK / what assumptions make it OK

Reply:
I give some reasoning slightly below if this is what you were thinking?

> Unicode characters only use 21 bits, so we can encode a trigram in 21 * 3 = 63 bits (a u64).